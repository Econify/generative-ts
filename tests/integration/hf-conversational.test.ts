import {
  createHuggingfaceInferenceModelProvider,
  HfConversationalTaskApi,
} from "@packages/core";

test("Huggingface - Conversational Task", async () => {
  // arrange
  const dialoGpt = createHuggingfaceInferenceModelProvider({
    api: HfConversationalTaskApi,
    modelId: "microsoft/DialoGPT-large",
  });

  // act
  const response = await dialoGpt.sendRequest({
    prompt: "What is the capital of France? Please respond in JSON format.",
    past_user_inputs: [
      "Whats the capital of Mexico? Please respond in JSON format.",
    ],
    generated_responses: [
      "{'answer': 'The capital of Mexico is Mexico City.'}",
    ],
  });

  // assert
  expect(response).toMatchApiSnapshot();
});
