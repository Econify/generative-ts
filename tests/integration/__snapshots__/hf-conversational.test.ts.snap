// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Huggingface - Conversational Task 1`] = `
[
  {
    "generated_text": "string",
  },
]
`;
